app/javascript/components/mentoring/DiscussionPostList.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect } from 'react'
+import { useRequestQuery } from '../../hooks/request-query'
+import { DiscussionPost, DiscussionPostProps } from './DiscussionPost'
+import { DiscussionPostChannel } from '../../channels/discussionPostChannel'
+
+export const DiscussionPostList = ({
+  endpoint,
+  discussionId,
+  iterationId,
+}: {
+  endpoint: string
+  discussionId: number
+  iterationId: number
+}): JSX.Element | null => {
+  const { isSuccess, data: posts, refetch } = useRequestQuery<
+    DiscussionPostProps[]
+  >(endpoint, { endpoint: endpoint, options: {} })
+
+  useEffect(() => {
+    const channel = new DiscussionPostChannel(
```

Comment:
When the websockets notification comes in, we tell `useRequestQuery` to refetch.